admin/class-yoast-form-select.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * @package WPSEO\Admin
+ */
+
+/**
+ * Class for generating a html select.
+ */
+class Yoast_Form_Select {
+
+	/**
+	 * @var string The field name for the select.
+	 */
+	private $field_name;
+
+	/**
+	 * @var string Complete name for the selects' name attribute.
+	 */
+	private $select_name;
+
+	/**
+	 * @var array Array with the options to parse.
+	 */
+	private $options;
+
+	/**
+	 * @var string The current selected option.
+	 */
+	private $selected_option;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $field_name      The field name for the select.
+	 * @param string $select_name     Complete name for the selects' name attribute.
+	 * @param array  $options         Array with the options to parse.
+	 * @param string $selected_option The current selected option.
+	 */
+	public function __construct( $field_name, $select_name, array $options, $selected_option ) {
+		$this->field_name      = $field_name;
+		$this->select_name     = $select_name;
+		$this->options         = $options;
+		$this->selected_option = $selected_option;
+	}
+
+	/**
+	 * Returns the generated html for the select.
+	 *
+	 * @return string
+	 */
+	public function get_html() {
+		$html = sprintf(
```

Comment:
Would like to have a Form Select HTML View which takes this object as input which handles the generating of the HTML. Which internally uses Form Option HTML Views to generate subparts for the options.